openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -0,0 +1,179 @@
+            `username` is not specified)
+        AccountNotAuthorized: the requesting_user does not have access to the user preference.
+        PreferenceNotFound: the user does not have a preference with the specified key.
+    """
+    existing_user = _get_user(requesting_user, username, allow_staff=True)
+    try:
+        preference_value = existing_user.preferences.get(key=preference_key).value
+    except ObjectDoesNotExist:
+        raise PreferenceNotFound()
+    return preference_value
+
+
+def get_user_preferences(requesting_user, username=None):
+    """Returns all user preferences as a JSON response.
+
+    Args:
+        requesting_user (User): The user requesting the user preferences. Only the user with username
+            `username` or users with "is_staff" privileges can access the preferences.
+        username (str): Optional username for which to look up the preferences. If not specified,
+            `requesting_user.username` is assumed.
+
+    Returns:
+         A dict containing account fields.
+
+    Raises:
+         AccountUserNotFound: no user with username `username` exists (or `requesting_user.username` if
+            `username` is not specified)
+        AccountNotAuthorized: the requesting_user does not have access to the user preference.
+    """
+    existing_user = _get_user(requesting_user, username, allow_staff=True)
+    user_serializer = UserSerializer(existing_user)
+    return user_serializer.data["preferences"]
+
+
+def update_user_preferences(requesting_user, update, username=None):
+    """Update the user preferences for the given username.
+
+    Note:
+        It is up to the caller of this method to enforce the contract that this method is only called
+        with the user who made the request.
+
+    Arguments:
+        requesting_user (User): The user requesting to modify account information. Only the user with username
+            'username' has permissions to modify account information.
+        update (dict): The updated account field values.
+        username (string): Optional username specifying which account should be updated. If not specified,
+            `requesting_user.username` is assumed.
+
+    Raises:
+        AccountUserNotFound: no user with username `username` exists (or `requesting_user.username` if
+            `username` is not specified)
+        AccountNotAuthorized: the requesting_user does not have access to change the account
+            associated with `username`
+        AccountUpdateError: the update could not be completed. Note that if multiple fields are updated at the same
+            time, some parts of the update may have been successful, even if an AccountUpdateError is returned;
+            in particular, the user account (not including e-mail address) may have successfully been updated,
+            but then the e-mail change request, which is processed last, may throw an error.
+    """
+    existing_user = _get_user(requesting_user, username)
```

Comment:
I thought it was cleaner to fetch the user up front, but then I refactored all the code into helper methods that can't take the user directly.